src/classes/action/link_action.vala

```diff
@@ -114,6 +114,33 @@ namespace pdfpc {
             );
         }
 
+        /**
+         * Launch an application, trying to cover all corner cases.
+         */
+        private void launch_for_uri(string uri) throws GLib.Error {
+            if (Uri.parse_scheme(uri) != null) {
+                AppInfo.launch_default_for_uri(uri, null);
+            } else {
+                // Guess a MIME type and launch its default handler
+                bool uncertain;
+                var ctype = ContentType.guess(uri, null, out uncertain);
+                var appinfo = AppInfo.get_default_for_type(ctype, false);
+                string path;
+                if (Path.is_absolute(uri)) {
+                    path = uri;
+                } else {
+                    // If the path is not absolute, translate it relative
+                    // to the PDF document location
+                    var pdf_fname = controller.get_pdf_fname();
+                    var dirname = Path.get_dirname(pdf_fname);
+                    path = Posix.realpath(Path.build_filename(dirname, uri));
```

Comment:
We do not officially support Windows, but is `Posix` working there? Can we use `Glib.File.new_build_filename()` instead?